[zkLogin] Replace prover endpoint with a link to Enoki (#17031)
MystenLabs/sui: docs/content/concepts/cryptography/zklogin.mdx

Because generating a ZK proof can be resource-intensive and potentially slow on the client side, it's advised that wallets utilize a backend service endpoint dedicated to ZK proof generation.

There are two options:
1. Call the Mysten Labs-maintained proving service.
1. Run the proving service in your backend using the provided Docker images.

### Call the Mysten Labs-maintained proving service

> If you wish to use the Mysten ran ZK Proving Service for Mainnet, please refer to [Enoki docs](https://docs.enoki.mystenlabs.com/) and contact us for accessing it.

Reviewer: ```suggestion
If you want to use the Mysten hosted ZK Proving Service for Mainnet, please refer to [Enoki docs](https://docs.enoki.mystenlabs.com/) and contact us for accessing it.
```

> The `prover-dev` endpoint (https://prover-dev.mystenlabs.com/v1) can be used freely for testing on Devnet. Note that proofs generated with this endpoint can only be submitted for Devnet zkLogin transactions, submitting it to Testnet or Mainnet fails.

Reviewer: ```suggestion
Use the `prover-dev` endpoint (https://prover-dev.mystenlabs.com/v1) freely for testing on Devnet. Note that you can submit proofs generated with this endpoint for Devnet zkLogin transactions only; submitting them to Testnet or Mainnet fails.
```
